### ISSUE TITLE
When `amount` is zero do not update the balance

### ISSUE THREAD

**Issue by xgreenx** [FuelLabs/fuel-vm#866]
The current code always increases and decreases the balance even if we have no updates. It means storage read and write and spends a lot of resources for nothing.

<img width="1046" alt="image" src="https://github.com/user-attachments/assets/b8cb9beb-51fd-4486-8b40-e5441b4c26da">

**Comment by askwhyharsh** [comment]
Interesting,

when is the increase/decrease happening? I would love to know more. As I can see, it updates a particular asset Id balance of a contract. It looks like a function, so I think it should be called when it's needed. So how is it calling even if we have no updates? Do you mean that if it's 0, we are still attempting an increase/decrease?

**Comment by xgreenx** [comment]
Yes, if value is zero, we still will update Merkel tree. So we need to do only update if values is not zero

**Comment by askwhyharsh** [comment]
i see, is it fine if i give it an attempt? can also try running the vm locally and play.
